lib/parsers/zip_parser/file_reader.rb

```diff
@@ -361,11 +360,13 @@ def locate_eocd_signature(in_str)
     minimum_record_size = 22
     end_location = minimum_record_size * -1
     loop do
+      # -100 here is an arbitrary number, but if not in place the this will continue
+      # reading gibberish (e.g. in our fuzz tests) forever
```

Comment:
This will blow up our fuzz tests if it doesn't have some way to fail past a certain end location. My solution (returning if the end location is less than -100) is a hacky lil patch - @julik, is there a way we can detect if something is a ZIP before hitting this loop? I've also tried setting this to be a ReadLimiter but it doesn't appear to make a difference, since ofc it's already done the reading by this point.

Reply:
@bermannoah I will make the loop bounded. Also, completely unrelated at all to this PR ;-)

Reply:
:-) No worries! Just wanted to make sure it wasn't the recursive decoder causing the issue.

Reply:
@julik would you like me to remove the -100 here since the fix is outside the scope of the PR? It does seem to cause issues with running the tests.